larq/optimizers_v2.py

```diff
@@ -1,10 +1,152 @@
+import warnings
+
 import tensorflow as tf
 import larq as lq
 
 from larq import utils
 from copy import deepcopy
 
 
+@utils.register_keras_custom_object
+class CaseOptimizer(tf.keras.optimizers.Optimizer):
+    """An optmizer wrapper that applies different optimizers to a subset of variables.
+
+    An optimizer is used to train a variable iff its accompanying predicate evaluates to
+    `True`.
+
+    For each variable, at most one optimizer's predicate may evaluate to `True`. If no
+    optimizer's predicate evaluates to `True` for a variable, it is trained with the
+    `default` optimizer.
```

Comment:
Maybe mention that if no `default` is set, the variable is not trained?

Reply:
Added in 35b3d36.